itou/files/management/commands/configure_bucket.py

```diff
@@ -79,8 +80,11 @@ def handle(self, *args, autoexpire=False, **options):
         )
 
     def check_minio(self):
+        # Response has a "400 bad request" status code, but we only care that the server is up.
+        current_logging_level = logging.getLogger("httpx").level
+        logging.getLogger("httpx").setLevel(logging.CRITICAL)
         response = httpx.head(settings.AWS_S3_ENDPOINT_URL)
```

Comment:
Merci pour la PR.
Une alternative serait d'appeler `urljoin(settings.AWS_S3_ENDPOINT_URL, "minio/health/live")` qui renvoie bien une 200 dans les cas où tout va bien (https://min.io/docs/minio/linux/operations/monitoring/healthcheck-probe.html#node-liveness).
On aurait toujours une ligne de log parasite mais avec une 200.

Reply:
Ah mais ce serait encore mieux effectivement, voire un mix des deux ? Utiliser l'url correcte, ET supprimer le log parasite ?
qu'en pensez-vous ?

Reply:
Il ne me semble pas nécessaire de retirer le log parasite. Si on souhaitait le faire, il serait prudent d’enrober le code qui change les niveaux de logs dans un `try: ... finally: ...`, pour éviter de changer la config des logs en cas d’exception. Ce qui commence à faire beaucoup de code pour éviter une ligne de log. La commande `configure_buckets` étant appelée dans les tests, un changement du niveau de log qui persiste pourrait avoir des conséquences désagréables.

Aussi, toute l’app (notamment `runserver`) affiche les requêtes émises par httpx, je ne vois pas de raison de traiter de manière spéciale cette commande.

Reply:
Si il y avait déjà ce helper dans le code, ce [context manager](https://docs.python.org/3/howto/logging-cookbook.html#using-a-context-manager-for-selective-logging) aurait été parfait pour l'occasion !

Effectivement, ça fait pas mal de code pour une fonctionnalité minime. Est-ce qu'on ferme cette PR ? Est-ce qu'on la change pour rajouter un `print("Le message d'erreur suivant est attendu, l'important est le code de retour de la commande")` ?

Ou alors... une modification du README pour préciser que ce message d'erreur n'est pas à prendre en compte, pour éviter la confusion chez ceux qui y seront confrontés pour la première fois ?

Reply:
Je pense que l’erreur n’est pas souhaitable, on garde la PR :+1:.
Je serais plutôt parti sur la solution de Xavier, changer l’URL pour `urljoin(settings.AWS_S3_ENDPOINT_URL, "minio/health/live")` afin d’avoir une 200 et non plus une 400.

Reply:
Sympa le context manager, peut-être qu’un jour... :grin: